src/Mod/Path/PathScripts/PathVcarve.py

```diff
@@ -0,0 +1,280 @@
+# *   GNU Library General Public License for more details.                  *
+# *                                                                         *
+# *   You should have received a copy of the GNU Library General Public     *
+# *   License along with this program; if not, write to the Free Software   *
+# *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  *
+# *   USA                                                                   *
+# *                                                                         *
+# ***************************************************************************
+
+import FreeCAD
+import Part
+import Path
+import PathScripts.PathEngraveBase as PathEngraveBase
+import PathScripts.PathLog as PathLog
+import PathScripts.PathOp as PathOp
+import PathScripts.PathUtils as PathUtils
+import PathScripts.PathGeom as PathGeom
+import PathScripts.PathPreferences as PathPreferences
+
+import traceback
+
+import math
+
+from PySide import QtCore
+
+__doc__ = "Class and implementation of Path Vcarve operation"
+
+
+if False:
+    PathLog.setLevel(PathLog.Level.DEBUG, PathLog.thisModule())
+    PathLog.trackModule(PathLog.thisModule())
+else:
+    PathLog.setLevel(PathLog.Level.INFO, PathLog.thisModule())
+
+
+# Qt tanslation handling
+def translate(context, text, disambig=None):
+    return QtCore.QCoreApplication.translate(context, text, disambig)
+
+
+class ObjectVcarve(PathEngraveBase.ObjectOp):
+    '''Proxy class for Vcarve operation.'''
+
+    def opFeatures(self, obj):
+        '''opFeatures(obj) ... return all standard features and edges based geomtries'''
+        return PathOp.FeatureTool | PathOp.FeatureHeights | PathOp.FeatureBaseFaces
+
+    def setupAdditionalProperties(self, obj):
+        if not hasattr(obj, 'BaseShapes'):
+            obj.addProperty("App::PropertyLinkList", "BaseShapes", "Path",
+                            QtCore.QT_TRANSLATE_NOOP("PathVcarve",
+                                "Additional base objects to be engraved"))
+        obj.setEditorMode('BaseShapes', 2)  # hide
+        if not hasattr(obj, 'BaseObject'):
+            obj.addProperty("App::PropertyLink", "BaseObject", "Path",
+                            QtCore.QT_TRANSLATE_NOOP("PathVcarve",
+                            "Additional base objects to be engraved"))
+        obj.setEditorMode('BaseObject', 2)  # hide
+
+    def initOperation(self, obj):
+        '''initOperation(obj) ... create vcarve specific properties.'''
+        obj.addProperty("App::PropertyFloat", "Discretize", "Path",
+                        QtCore.QT_TRANSLATE_NOOP("PathVcarve",
+                        "The deflection value for discretizing arcs"))
+        obj.addProperty("App::PropertyFloat", "Threshold", "Path",
+                        QtCore.QT_TRANSLATE_NOOP("PathVcarve",
+                        "cutoff for removing colinear segments (degrees). \
+                        default=10.0."))
```

Comment:
The threshold is a bit mangled. In the UI it says it's a value between 0 and 1 and the default is 0.8. The actual default value is 1 though, due to it being set to 10 here (I presume the spin box has a limit set).

The value is used directly for the colinear filtering, which requires an angle as input. The idea of abstracting it into a scale is interesting. Not sure if there is a use case where you wanted to eliminate ribs ending in a 90° (or more) corner. What were your thoughts on the upper limit?

Reply:
fixed.  Attribute now defaults to filtering colinear segments within 10 degrees.  The control takes an integer between 0 - 180 degrees